Automattic/simplenote-ios
# In app account deletion

### Fix
This PR Fixes #1353 

Per upcoming changes to the app store use guidelines we need to add In app account deletion.  This PR adds this feature to SNiOS

<img src="https://cdn-std.droplr.net/files/acc_593859/HyETT8" />

Note that currently the end point we are using to trigger deletion is in progress so a temporary url was added for testing purposes.

Note that currently the PR is missing some details needed to make it run.  Ping me for the details needed to make it compile
Discussion here p1627443887039600-slack-C01KRRK0MLJ
### Test
1. Log into simplenote to an account you don't mind deleting
2. Go to Settings -> Delete Account
3. Tap on the delete account button, confirm deletion
4. After a few seconds you should see a confirmation button.  Click on Okay
5. Check the email associated with the account you should receive a verification email to confirm deleting.

Because of the changes to Remote and Result, also test signing up to Simplenote from the app to ensure the SignupRemote still operates correctly

### Review
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in 37db49 with:
> 
> > -   Added In app account deletion #1354
